src/components/dictionaryConcepts/components/CreateMapping.jsx

```diff
@@ -2,26 +2,55 @@ import React, { Component } from 'react';
 import AsyncSelect from 'react-select/lib/Async';
 import PropTypes from 'prop-types';
 import { fetchSourceConcepts } from '../../../redux/actions/concepts/dictionaryConcepts';
-import { INTERNAL_MAPPING_DEFAULT_SOURCE, CIEL_SOURCE_URL } from './helperFunction';
+import { INTERNAL_MAPPING_DEFAULT_SOURCE, CIEL_SOURCE_URL, MAP_TYPES_DEFAULTS } from './helperFunction';
 import MapType from './MapType';
 
 class CreateMapping extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      type: '',
+      editMapType: '',
+    };
   }
 
   handleInputChange = (value) => {
     this.setState({ inputValue: value });
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps) {
+      this.setState({
+        type: nextProps.map_type,
+      });
+    }
+  }
+
+  componentWillMount() {
+    const editMapType = this.props.source;
+    this.setState({
+      editMapType,
+    });
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.source !== this.props.source) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({
+        type: '',
+      });
+    }
+    console.log('Hit CDU');
```

Comment:
Kindly remove the console.log statemnet

Reply:
The console.log statements are there purely for testing purposes. When I bulk up my test coverage, I will remove them